module/VuFind/src/VuFind/Content/Covers/Bokinfo.php

```diff
@@ -0,0 +1,200 @@
+ * @package  Content
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Bokinfo extends \VuFind\Content\AbstractCover implements
+    \Laminas\Log\LoggerAwareInterface,
+    \VuFindHttp\HttpServiceAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->supportsIsbn = $this->cacheAllowed = true;
+    }
+
+    /**
+     * Get image URL for a particular API key and set of IDs (or false if invalid).
+     *
+     * @param string $key  API key
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        if (!isset($ids['isbn'])) {
+            return false;
+        }
+        if (!isset($key)) {
+            return false;
+        }
+
+        $isbn = $ids['isbn']->get13();
+        try {
+            $client = $this->createHttpClient(
+                "https://api.bokinfo.se/book/get/$isbn"
+            );
+
+            $client->getRequest()->getHeaders()
+                  ->addHeaderLine("Ocp-Apim-Subscription-Key", $key);
+
+            try {
+                $resp= $client->send();
+                $body = $resp->getBody();
+                $url = $this->getImageUrl($body);
+            } catch (\Exception $e) {
+            }
+
+            if ($this->testUrlFunction($url)) {
+                return "$url";
+            }
+        } catch (\Throwable $ex) {
+            return false;
+        } catch (\Exception $ex) {
+            return false;
+        } catch (\RuntimeException $ex) {
+            return false;
+        }
+
+        return false;
+    }
+
+    /**
+     * Return a HTTP Client object
+     *
+     * @param string $url API Url
+     *
+     * @return HttpClient Http Client
+     */
+    protected function createHttpClient($url)
+    {
+        $client = $this->httpService->createClient($url);
+
+        $adapter = $client->getAdapter();
+        if ($adapter instanceof \Laminas\Http\Client\Adapter\Socket) {
+            $context = $adapter->getStreamContext();
+            $res = stream_context_set_option(
+                $context,
+                'ssl',
+                'verify_peer_name',
+                false
+            );
+            if (!$res) {
+                throw new \Exception('Unable to set sslverifypeername option');
+            }
+        } elseif ($adapter instanceof \Laminas\Http\Client\Adapter\Curl) {
+            $adapter->setCurlOption(CURLOPT_SSL_VERIFYHOST, false);
+        }
+
+        // Set timeout value
+        $timeout = 30;
```

Comment:
It might be useful to make the timeout value a class property rather than a local variable -- then it's easier to customize or make configurable in future if a need arises.

Reply:
Actually, it might not be needed If I leave it out it chould take the \VuFindHttp\HttpServiceAwareInterface default right?

Reply:
Even if you take this out, you'll still need `HttpServiceAwareInterface` -- that interface is a marker that tells the framework to inject the service that creates HTTP clients, and you need that for the `createHttpClient` method to work. Of course, if you prefer not to rely on this kind of "magic," you could also set up a custom factory for the plug-in and inject the client via the factory.